src/main/java/org/uwdigi/rag/service/SqlDatabaseContentRetriever.java

```diff
@@ -135,24 +137,33 @@ public SqlDatabaseContentRetriever(
       DataSource dataSource,
       String sqlDialect,
       String databaseStructure,
+      String sqlPromptTemplate,
       PromptTemplate promptTemplate,
       ChatLanguageModel chatLanguageModel,
       @Qualifier("ollamaChatLanguageModel") ChatLanguageModel ollamaChatModel,
       AssistantService assistantService,
       FhirDbConfig fhirDbConfig,
       Map<String, String> tables,
+      boolean useCloudLLMOnly,
       EmbeddingStore<TextSegment> embeddingStore,
       EmbeddingModel embeddingModel,
       boolean setUp,
       Integer maxRetries) {
     this.dataSource = ensureNotNull(dataSource, "dataSource");
     this.sqlDialect = getOrDefault(sqlDialect, () -> getSqlDialect(dataSource));
     this.databaseStructure = getOrDefault(databaseStructure, () -> generateDDL(dataSource));
-    this.promptTemplate = getOrDefault(promptTemplate, DEFAULT_PROMPT_TEMPLATE);
```

Comment:
So we would just use this method. which I believe handles the null point of 'promptTemplate'

Reply:
thanks though that is being handles in the change introduced below